alf/trainers/evaluator.py

```diff
@@ -267,6 +267,7 @@ def evaluate(env: AlfEnvironment, algorithm: RLAlgorithm,
     """
     batch_size = env.batch_size
     env.reset()
+    env.sync_progress()
```

Comment:
This has to be qualified so that only parallel_envs are allowed, otherwise sync_progress() from a thread env will throw RuntimeError?

Reply:
Now the default implementation of sync_progress() does not fail.